test/flux_calculator_tests.jl

```diff
@@ -50,3 +63,222 @@ end
+    (; F_turb_energy) = fields
+    @. sim.integrator.p.F_aero[colidx] = F_turb_energy
+end
+
+@testset "calculate correct fluxes: dry" begin
+    surface_scheme_list = (MoninObukhovScheme(), BulkScheme())
+    for scheme in surface_scheme_list
+        boundary_space = TestHelper.create_space(FT)
+
+        params = (; surface_scheme = scheme, FT = FT)
+
+        # atmos
+        p = (;
+            energy_bc = zeros(boundary_space),
+            ρq_tot_bc = zeros(boundary_space),
+            uₕ_bc = ones(boundary_space),
+            z = ones(boundary_space),
+            z_sfc = zeros(boundary_space),
+            u = ones(boundary_space),
+            v = ones(boundary_space),
+        )
+        Y_init = (; ρ = ones(boundary_space) .* 1.2, T = ones(boundary_space) .* 310, q = zeros(boundary_space))
+        integrator = (; Y_init..., p = p)
+        atmos_sim = TestAtmos(params, Y_init, nothing, integrator)
+
+        # ocean
+        p = (;
+            F_aero = zeros(boundary_space),
+            z0m = FT(0.01),
+            z0b = FT(0.01),
+            beta = ones(boundary_space),
+            α = ones(boundary_space) .* 0.5,
+            q = zeros(boundary_space),
+            Cd = FT(0.01),
+            Ch = FT(0.01),
+            area_fraction = ones(boundary_space) .* 0.5,
+        )
+        Y_init = (; T = ones(boundary_space) .* 300.0)
+        integrator = (; Y_init..., p = p)
+        ocean_sim = TestOcean(nothing, Y_init, nothing, integrator)
+
+        # ocean
+        ocean_sim2 = TestOcean(nothing, Y_init, nothing, integrator)
+
+        model_sims = (; atmos_sim, ocean_sim, ocean_sim2)
+
+        coupler_cache_names = (
+            :T_S,
+            :albedo,
+            :F_R_sfc,
+            :F_R_toa,
+            :P_liq,
+            :P_snow,
+            :P_net,
+            :F_turb_energy,
+            :F_turb_ρτxz,
+            :F_turb_ρτyz,
+            :F_turb_moisture,
+        )
+        fields = NamedTuple{coupler_cache_names}(ntuple(i -> Fields.zeros(boundary_space), length(coupler_cache_names)))
+
+        # calculate turbulent fluxes
+        thermo_params = get_thermo_params(atmos_sim)
+        partitioned_turbulent_fluxes!(model_sims, fields, boundary_space, scheme, thermo_params)
+
+        # calculating the fluxes twice ensures that no accumulation occurred
+        partitioned_turbulent_fluxes!(model_sims, fields, boundary_space, scheme, thermo_params)
+
+        windspeed = @. hypot(atmos_sim.integrator.p.u, atmos_sim.integrator.p.v)
+
+        thermo_params = get_thermo_params(atmos_sim)
+        thermo_state_int = Interfacer.get_field(atmos_sim, Val(:thermo_state_int))
+
+        surface_thermo_states = similar(thermo_state_int)
+        Fields.bycolumn(boundary_space) do colidx
+            surface_thermo_states[colidx] .=
+                surface_thermo_state(ocean_sim, thermo_params, thermo_state_int[colidx], colidx)
+        end
+
+        if scheme == BulkScheme()
```

Comment:
Could you add a comment explaining what we do differently for a `BulkScheme`? Also, should this use `isa` instead of `==`?